users/engler/fairseq/training.py

```diff
@@ -76,26 +101,37 @@ def __init__(
             self.keep_epochs = set(keep_epochs)
         self.max_epoch = max_epoch
         self.save_interval = save_interval
+        self.fairseq_python_exe = (
+            fairseq_python_exe
+            if fairseq_python_exe is not None
+            else gs.FAIRSEQ_PYTHON_EXE
+        )
+        self.fairseq_hydra_exe = (
+            fairseq_hydra_exe if fairseq_hydra_exe is not None else gs.FAIRSEQ_HYDRA_EXE
+        )
+        self.fairseq_root = fairseq_root
```

Comment:
I think we should assert that either of `fairseq_hydra_exe` and `fairseq_root` is `None` because then we clearly know which version to use in `_get_run_cmd()`.

Reply:
Yes exactly, it should be XOR.